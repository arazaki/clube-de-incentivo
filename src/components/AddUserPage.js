import React from 'react';
import { connect } from 'react-redux';
import UserForm from './UserForm';
import { startAddUser } from '../actions/user';
import { startLogout } from '../actions/auth';

export class AddUserPage extends React.Component {
    onSubmit = (user) => {
        this.props.startAddUser(user);
        this.props.history.push('/');
    };

    onLogout = () => {
        this.props.startLogout();
    }

    render() {
        return (
            <div>
                <div className="header">
                    <div className="content-container">
                        <div className="header__content">
                            <h1 className="header__title">Criar usu√°rio</h1>
                            <button className="button button--link" onClick={this.onLogout}>Sair</button>
                        </div>
                    </div>
                </div>
                <div className="content-container content-container--form">
                    <UserForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddUser: (user) => dispatch(startAddUser(user)),
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(AddUserPage);