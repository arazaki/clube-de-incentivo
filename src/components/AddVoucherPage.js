import React from 'react';
import { connect } from 'react-redux';
import VoucherForm from './VoucherForm';
import { startAddVoucher } from '../actions/vouchers';

export class AddVoucherPage extends React.Component {
    onSubmit = (voucher) => {
        this.props.startAddVoucher(voucher);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Resgatar</h1>
                    </div>
                </div>
                <div className="content-container">
                    <VoucherForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddVoucher: (voucher) => dispatch(startAddVoucher(voucher))
});

export default connect(undefined, mapDispatchToProps)(AddVoucherPage);