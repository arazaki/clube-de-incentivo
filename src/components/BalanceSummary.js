import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import numeral from 'numeral';

const BalanceSummary = ({ balance, companyName, id }) => {
    const formattedBalanceTotal = numeral(balance).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <div className="summary-content">
                    <div className="summary-balance">
                        <h1 className="NOCLASS">{companyName}</h1>
                        <h1 className="page-header__title">
                            Saldo dispon√≠vel: <span>{formattedBalanceTotal}</span>
                        </h1>
                        <div className="nav-link">
                            <NavLink className="nav-link__link" to={`/dashboard/${id}`} activeClassName="is-active">Extrato</NavLink>
                            <NavLink className="nav-link__link" to={`/purchased/${id}`} activeClassName="is-active">Compras</NavLink>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default BalanceSummary;