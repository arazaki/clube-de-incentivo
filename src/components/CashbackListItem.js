import React from 'react';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ clientDocument, total, purchaseDate, items, cfeId }) => (
    <div className="list-item">
        <div>
            <h3 className="list-item__title">{moment(purchaseDate).add(7, 'days').format('DD [de] MMMM [de] YYYY')}</h3>
        </div>
        <div>
            <h3 className="list-item__data">
                Recebe de volta: {numeral(total * 0.1).format('$0,0.00')}
            </h3>
        </div>
    </div>
);

export default ExpenseListItem;