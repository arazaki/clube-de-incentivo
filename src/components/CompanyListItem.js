import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const CompanyListItem = ({ id, companyName, balance }) => (
    <Link className="NOCLASS" to={`/dashboard/${id}`}>
        <div className="list-item">
            <div>
                <h3 className="list-item__title">{companyName}</h3>
                <h3 className="list-item__data">
                    Saldo: {numeral(balance).format('$0,0.00')}
                </h3>
            </div>
        </div>
    </Link>
);

export default CompanyListItem;