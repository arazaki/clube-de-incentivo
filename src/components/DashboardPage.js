import React from 'react';
import { connect } from 'react-redux';
import StatementList from './StatementList';
import BalanceSummary from './BalanceSummary';

export const DashboardPage = ({ company }) => (
    <div className="bg-orange">
        <BalanceSummary {...company} />
        <StatementList {...company} />
    </div>
);

const mapStateToProps = (state, props) => ({
    company: state.companies.find((company) => company.id === props.match.params.id)

});

export default connect(mapStateToProps)(DashboardPage);


