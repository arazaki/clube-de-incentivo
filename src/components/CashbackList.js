import React from 'react';
import { connect } from 'react-redux';
import CashbackListItem from './CashbackListItem';
import selectExpenses from '../selectors/expenses';

export const CashbackList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Data</div>
            <div className="show-for-desktop">Cashbacks</div>
            <div className="show-for-desktop">Cashback</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>
                            Sem compras ainda :/
                    </span>
                    </div>
                ) : (
                        props.expenses.map((expense) => {
                            return <CashbackListItem key={expense.id} {...expense} />;
                        })
                    )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(CashbackList);