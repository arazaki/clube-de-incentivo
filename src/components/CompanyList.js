import React from 'react';
import { connect } from 'react-redux';
import CompanyListItem from './CompanyListItem';

export const CompanyList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="NOCLASS">Clubes</div>
        </div>
        <div className="list-body">
            {
                props.companies.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>
                            Sem clubes ainda :/
                    </span>
                    </div>
                ) : (
                        props.companies.map((company) => {
                            return <CompanyListItem key={company.id} {...company} />;
                        })
                    )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        companies: state.companies
    };
};

export default connect(mapStateToProps)(CompanyList);