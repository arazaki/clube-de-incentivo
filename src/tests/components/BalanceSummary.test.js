import React from 'react';
import { shallow } from 'enzyme';
import { BalanceSummary } from '../../components/BalanceSummary';

test('should correctly render BalanceSummary with 1 expense', () => {
    const wrapper = shallow(<BalanceSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render BalanceSummary with multiple expenses', () => {
    const wrapper = shallow(<BalanceSummary expenseCount={23} expensesTotal={23512340987} />);
    expect(wrapper).toMatchSnapshot();
});