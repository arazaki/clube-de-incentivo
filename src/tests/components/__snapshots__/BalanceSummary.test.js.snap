// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render BalanceSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Saldo disponível: 
      <span>
        $0.00
      </span>
    </h1>
  </div>
  <div
    className="nav-link"
  >
    <div
      className="content-container"
    >
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="nav-link__link"
        to="/dashboard"
      >
        Extrato
      </NavLink>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="nav-link__link"
        to="/futureCashback"
      >
        Futuro
      </NavLink>
    </div>
  </div>
</div>
`;

exports[`should correctly render BalanceSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Saldo disponível: 
      <span>
        $0.00
      </span>
    </h1>
  </div>
  <div
    className="nav-link"
  >
    <div
      className="content-container"
    >
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="nav-link__link"
        to="/dashboard"
      >
        Extrato
      </NavLink>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        className="nav-link__link"
        to="/futureCashback"
      >
        Futuro
      </NavLink>
    </div>
  </div>
</div>
`;
